### ISSUE TITLE
[Usage]: 使用Vllm如何对Qwen3ForSequenceClassification模型进行文本分类加速？

### ISSUE THREAD

**Issue by Alex-DeepL** [vllm-project/vllm#19950]
### Your current environment

```text
The output of `python collect_env.py`
```
from fastapi import FastAPI
from transformers import AutoTokenizer
from pydantic import BaseModel
import torch
import asyncio
from vllm import LLM
import os
import uvicorn

class cls_request(BaseModel):
    text: str

os.environ["CUDA_VISIBLE_DEVICES"] = '0'
label_list = ['家用电器', '机械设备', '建筑材料', '电力设备', '商贸零售', '国防军工', '医药生物', '建筑装饰', '电子',
              '传媒', '基础化工', '美容护理', '社会服务', '计算机', '环保', '汽车', '食品饮料', '轻工制造', '有色金属',
              '交通运输', '农林牧渔', '公用事业', '房地产', '石油石化', '银行', '钢铁', '通信', '非银金融', '纺织服饰',
              '煤炭']

app = FastAPI()

merged_model_path = "/home/lwl/wordkspace/LLaMA-Factory/Qwen3Classification/checkpoint/merged_model"
cls_model = LLM(model=merged_model_path, trust_remote_code=True, dtype='bfloat16', task="classify",max_model_len=2048)
cls_tokenizer = AutoTokenizer.from_pretrained(merged_model_path)
# 队列和批量处理参数
request_queue = []
BATCH_SIZE = 10
BATCH_TIMEOUT = 0.01  # 秒

@app.on_event("startup")
async def startup_event():
    asyncio.create_task(batch_worker())

async def batch_worker():
    while True:
        await asyncio.sleep(BATCH_TIMEOUT)
        if len(request_queue) == 0:
            continue
        batch = []
        while len(batch) < BATCH_SIZE and request_queue:
            batch.append(request_queue.pop(0))
        if batch:
            texts_inputs = [cls_tokenizer.decode(cls_tokenizer(item['text'],truncation=True, max_length=2048)['input_ids']) for item in batch]
            with torch.no_grad():
                outputs = cls_model.classify(texts_inputs)
                predictions = [item.outputs.probs.index(max(item.outputs.probs)) for item in outputs]
            for item, pred in zip(batch, predictions):
                item['future'].set_result(label_list[pred])

@app.post("/cls")
async def get_cls(cls: cls_request):
    loop = asyncio.get_event_loop()
    future = loop.create_future()
    request_queue.append({'text': cls.text, 'future': future})
    label = await future
    return {"label": label}

if __name__ == '__main__':
    uvicorn.run(app, host="0.0.0.0", port=18010)

### How would you like to use vllm

Currently, I am using LLM in vllm for service deployment. Now I want to use AsyncLLMEngine for optimization. How should I make the modifications

### Before submitting a new issue...

- [x] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.

**Comment by noooop** [comment]
try: 

vllm serve jason9693/Qwen2.5-1.5B-apeach --task classify


you may need dev version of vllm

https://docs.vllm.ai/en/latest/getting_started/installation/gpu.html#install-the-latest-code_1

**Comment by whyiug** [comment]
Same question, seems like prefix cache doesn't work.

**Comment by Alex-DeepL** [comment]
use the latest vllm is ok, install vllm from source code





------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "vllm-project/vllm"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2025年7月1日(星期二) 下午3:19
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [vllm-project/vllm] [Usage]: 使用Vllm如何对Qwen3ForSequenceClassification模型进行文本分类加速？ (Issue #19950)



whyiug left a comment (vllm-project/vllm#19950)
 
Same question, seems like prefix cache doesn't work.

**Comment by whyiug** [comment]
> use the latest vllm is ok, install vllm from source code
> […](#)

no, I can deploy it with 'vllm serve jason9693/Qwen2.5-1.5B-apeach --task classify

', it's just that the prefix cache doesn't work.

**Comment by umie0128** [comment]
@DarkLight1337  请问现在支持这个吗？

**Comment by DarkLight1337** [comment]
This model should be supported

**Comment by noooop** [comment]
try

```
model = LLM(
    model="",
    task="classify",
    enable_prefix_caching=True,
    override_pooler_config={"pooling_type": "LAST"}
)
```

or wait for #20930 landing, which wait for #20538 landing

**Comment by umie0128** [comment]
> try
> 
> ```
> model = LLM(
>     model="",
>     task="classify",
>     enable_prefix_caching=True,
>     override_pooler_config={"pooling_type": "LAST"}
> )
> ```
> 
> or wait for [#20930](https://github.com/vllm-project/vllm/pull/20930) landing, which wait for [#20538](https://github.com/vllm-project/vllm/pull/20538) landing

请问这个适配的模型是从哪里获取，我用modelscope/ms-swift lora微调 --task_type seq_cls 训练后merge得到微调后的模型，再用 swift infer --infer_backend vllm 会报AttributeError: 'Qwen3ForSequenceClassification' object has no attribute 'logits_processor'。这个主要是transformers的Qwen3ForSequenceClassification是给qwen加个检测头，但是在vllm是专门为qwen3-rerank服务，后续这个函数会改动吗？
